subsys/net/ip/net_if.c

```diff
@@ -4365,6 +4433,169 @@ bool net_if_is_promisc(struct net_if *iface)
 	return net_if_flag_is_set(iface, NET_IF_PROMISC);
 }
 
+int net_if_set_opt(struct net_if *iface, int optname, const void *optval, size_t optlen)
+{
+	int status = 0;
+
+	k_mutex_lock(&lock, K_FOREVER);
+
+	const struct net_if_api *api = net_if_get_device(iface)->api;
+
+	if (!(api->set_opt)) {
+		status = -ENOTSUP;
+		goto out;
+	}
+
+	/* Question for reviewers:
+	 *
+	 * I am not sure whether it would be cleaner to:
+	 *
+	 * 1) Let each iface that supports either NET_IF_OPT_CONNECTIVITY_PERSISTENT or
+	 *    NET_IF_OPT_CONNECTIVITY_TIMEOUT implement those options themselves, OR:
+	 *
+	 * 2) Create a global implementation for both options in net_if.c, and require
+	 *    ifaces that support the options call net_if_get_opt on themselves to check the
+	 *    current value.
+	 *
+	 * Either approach has its pros and cons. What do the reviewers think?
+	 */
+
+
+	/* Related question:
+	 *
+	 * If we decide to create a global implementation for the options, we could even go so far
+	 * as to implement the timeout behavior directly in net_if.c as a timer that calls, say,
+	 * "net_if_api->handle_timeout" on the iface.
+	 *
+	 * This runs the risk of dipping too deeply into the inner-workings of network ifaces,
+	 * though. I lean more towards letting each iface implement its timeout mechanism how
+	 * it sees fit.
+	 */
+
+	status = api->set_opt(optname, optval, optlen);
+
+out:
+	k_mutex_unlock(&lock);
+
+	return status;
+}
+
+int net_if_get_opt(struct net_if *iface, int optname, const void *optval, size_t *optlen)
+{
+	int status = 0;
+
+	k_mutex_lock(&lock, K_FOREVER);
+
+	const struct net_if_api *api = net_if_get_device(iface)->api;
+
+	if (!(api->get_opt)) {
+		status = -ENOTSUP;
+		goto out;
+	}
+
+	status = api->get_opt(optname, optval, optlen);
+
+out:
+	k_mutex_unlock(&lock);
+
+	return status;
+}
+
+
+/* Note to reviewers:
+ * These helper functions were not discussed in the RFC.
+ * After writing an implementation of conn_mgr, I thought some helper functions for the
+ * global options would be nice.
+ *
+ * What do you think?
```

Comment:
I'd keep the net_if API small. You can move these as statics to `conn_mgr` for now.